### PR TITLE
Same fendermint docker image to contain both fendermint and ipc-cli

### PR DESCRIPTION
**Pull request by mb1896**
Closes ENG-677

In this PR, we put `ipc-cli` in the fendermint docker image.

We can keep using `docker run` for fendermint images like previously. At the same time, we can also run
`docker run <image> ipc-cli` to access to `ipc-cli` and its subcommands.

### CODE REVIEW

**Review by raulk — COMMENTED** [review]
This is missing the entrypoint script to dispatch to the Fendermint or ipc-cli executable. It is also missing the push to both the fendermint and ipc image repos in from the GitHub workflow.

**Comment by raulk** [comment]
Can you please add a description?

**Comment by mb1896** [comment]
> Referen

Done.

**Comment by mb1896** [comment]
> This is missing the entrypoint script to dispatch to the Fendermint or ipc-cli executable. It is also missing the push to both the fendermint and ipc image repos in from the GitHub workflow.

Fixed as you suggested.

**Comment by mb1896** [comment]
> This is missing the entrypoint script to dispatch to the Fendermint or ipc-cli executable. It is also missing the push to both the fendermint and ipc image repos in from the GitHub workflow.

There is only one fendermint images, not two images. So all existing auto-push workflow should work.

File: fendermint/docker/builder.ci.Dockerfile [hunk 1]
```diff
@@ -70,7 +70,8 @@ RUN set -eux; \
   *) echo >&2 "unsupported architecture: ${TARGETARCH}"; exit 1 ;; \
   esac; \
   rustup show ; \
-  cargo build --release -p fendermint_app --target ${ARCH}-unknown-linux-gnu
+  cargo build --release -p fendermint_app --target ${ARCH}-unknown-linux-gnu &&\
+  cargo build --release -p ipc-cli --target ${ARCH}-unknown-linux-gnu
```
Comment (aakoshh) [inline fendermint/docker/builder.ci.Dockerfile:74]: Let's add `--locked` here too. I added in one of my draft PRs. I think if the `cargo install` below has it, best if this one does too.

File: fendermint/docker/docker_entry.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# This script should be used as ENTRYPOINT for fendermint docker image.
+
+CMD=$1
+
+if [[ $CMD == 'ipc-cli' ]]; then
+  /usr/local/bin/ipc-cli "${@:2}"
```
Comment (aakoshh) [inline fendermint/docker/docker_entry.sh:8]:
```suggestion
  exec ipc-cli "${@:2}"
```
So it handled signals.

File: fendermint/docker/docker_entry.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+else
+  if (( $# == 0)); then
+    /usr/local/bin/fendermint run
```
Comment (aakoshh) [inline fendermint/docker/docker_entry.sh:11]:
```suggestion
    exec fendermint run
```

File: fendermint/docker/docker_entry.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  else
+    /usr/local/bin/fendermint "$@"
```
Comment (aakoshh) [inline fendermint/docker/docker_entry.sh:13]:
```suggestion
    exec fendermint "$@"
```

File: fendermint/docker/runner.Dockerfile [hunk 5]
```diff
@@ -30,5 +29,7 @@ RUN chmod 777 /fendermint/logs
 COPY fendermint/docker/.artifacts/bundle.car $FM_HOME_DIR/bundle.car
 COPY fendermint/docker/.artifacts/custom_actors_bundle.car $FM_HOME_DIR/custom_actors_bundle.car
 COPY fendermint/docker/.artifacts/contracts $FM_HOME_DIR/contracts
+COPY fendermint/docker/docker_entry.sh /usr/local/bin/docker_entry.sh
```
Comment (aakoshh) [inline fendermint/docker/runner.Dockerfile:32]: nit: I think kebab-casing is more idiomatic with scripts, ie. `docker-entry.sh`

File: fendermint/docker/docker_entry.sh (hunk 3 above)
Comment (aakoshh) [reply fendermint/docker/docker_entry.sh:11]: I wonder if this case is even possible, with the dockerfile specifying `"run"` as the default `CMD`.

File: fendermint/docker/docker_entry.sh (hunk 2 above)
Comment (mb1896) [reply fendermint/docker/docker_entry.sh:8]: Yeah the `$@` is the array that contains all argument for this script.
`${@:2}` means its subarray starting from the second element.
The double quote is not compsiultory but recommended , see https://github.com/koalaman/shellcheck/wiki/SC2068.

File: fendermint/docker/builder.ci.Dockerfile (hunk 1 above)
Comment (mb1896) [reply fendermint/docker/builder.ci.Dockerfile]: Added. Thanks!

File: fendermint/docker/docker_entry.sh (hunk 3 above)
Comment (mb1896) [reply fendermint/docker/docker_entry.sh]: Yes it's possible! Tested. Thanks!

File: fendermint/docker/docker_entry.sh (hunk 4 above)
Comment (mb1896) [reply fendermint/docker/docker_entry.sh]: Fixed.

File: fendermint/docker/runner.Dockerfile (hunk 5 above)
Comment (mb1896) [reply fendermint/docker/runner.Dockerfile]: Sure, fixed!

**Comment by mb1896** [comment]
@aakoshh @raulk Can you review this PR when you get a chance? Thanks!

**Review by aakoshh — APPROVED** [review]
Sweet!

**Pull request merged**